File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
int bsl(vector<int> arr,int target){
    int l=0;
    int h=arr.size()-1;
    int ans=-1;
    while(l<=h){
        int mid=l+(h-l)/2;
        if(arr[mid]==target){
            ans=mid;
            h=mid-1;
        }
        else if(arr[mid]>target)h=mid-1;
        else l=mid+1;
    }
    return ans;
}

int bsr(vector<int>arr,int target){
    int l=0;
    int h=arr.size()-1;
    int ans=-1;
    while(l<=h){
        int mid=l+(h-l)/2;
        if(arr[mid]==target){
            ans=mid;
            l=mid+1;
        }
        else if(arr[mid]>target)h=mid-1;
        else l=mid+1;
    }
    return ans;
}
    vector<int> searchRange(vector<int>& arr, int target) {
        vector<int>ans;
        int fr=bsl(arr,target);
        int ls=bsr(arr,target);
        ans.push_back(fr);
        ans.push_back(ls);
        return ans;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int left(int[] arr, int target){
        int ans = -1;
        int l = 0;
        int r = arr.length-1;
        while(l<=r){
            int mid  = l + (r - l)/2;
            if(arr[mid]==target){
                ans = mid;
                r = mid -1; 
            }
            else if(arr[mid]>target)r=mid-1;
            else l=mid+1;
        }
        return ans;
    }

    public int right(int[] arr, int target){
        int ans  = -1;
        int l = 0, r = arr.length-1;
        while(l<=r){
            int mid = l + (r-l)/2;
            if(arr[mid]==target){
                ans = mid;
                l=mid+1;
            }
            else if(arr[mid]>target)r=mid-1;
            else l = mid + 1;
        }
        return ans; 
    }
    public int[] searchRange(int[] arr, int target) {
        int lans = left(arr, target);
        int rans = right(arr, target);
        int[] ans = {lans,rans};
        return ans;
    }
}